src/pages/publication/PublicationForm.tsx

```diff
@@ -67,47 +67,46 @@ const PublicationForm: FC<PublicationFormProps> = ({
   const dispatch = useDispatch();
   const history = useHistory();
 
-  useEffect(() => {
-    // Get files uploaded from new publication view
-    const files = Object.values(uppy.getState().files).map((file) => ({
-      ...emptyFile,
-      identifier: file.id,
-      name: file.name,
-      mimeType: file.type ?? '',
-      size: file.size,
-    }));
-
-    if (files?.length) {
-      setInitialValues({
-        ...emptyPublication,
-        fileSet: { ...emptyFileSet, files },
-      });
-    }
-  }, [uppy]);
-
   useEffect(() => {
     return () => uppy && uppy.reset();
   }, [uppy]);
 
   useEffect(() => {
     const getPublicationById = async (id: string) => {
+      const files = Object.values(uppy.getState().files).map((file) => ({
+        ...emptyFile,
+        identifier: file.response?.uploadURL ?? file.id,
+        name: file.name,
+        mimeType: file.type ?? '',
+        size: file.size,
+      }));
+
       const publication = await getPublication(id);
+
       if (publication.error) {
         closeForm();
         dispatch(setNotification(publication.error, NotificationVariant.Error));
       } else if (publication.status === PublicationStatus.PUBLISHED) {
         history.push(`/publication/${id}/public`);
       } else {
         // TODO: revisit necessity of deepmerge when backend model has all fields
-        setInitialValues(deepmerge(emptyPublication, publication));
+        setInitialValues(
+          deepmerge(
+            {
+              ...emptyPublication,
+              fileSet: { ...emptyFileSet, files },
```

Comment:
dette betyr at om man laster opp fil, starter registrering, trykker F5, så er fila borte?
jeg ville forventa at minst ei fil alltid er en del av publikasjonen når man starter registrering med fil

Reply:
ja, men det vil jo skje uansett sånn som det funker med uppy nå. hva mener du at vi skal gjøre for å unngå det?

Reply:
lagre med en gang den er opprettet?

Reply:
det jeg tenkte ja
opprett tom -> lagre med fil -> åpne form

Reply:
ok, ja - det er lurt 👌